pkg/aggregator/sender.go

```diff
@@ -295,11 +301,31 @@ func (s *checkSender) HistogramBucket(metric string, value int64, lowerBound, up
 // Historate should be used to create a histogram metric for "rate" like metrics.
 // Warning this doesn't use the harmonic mean, beware of what it means when using it.
 func (s *checkSender) Historate(metric string, value float64, hostname string, tags []string) {
-	s.sendMetricSample(metric, value, hostname, tags, metrics.HistorateType, false, false)
+	s.sendMetricSample(metric, value, hostname, tags, metrics.HistorateType, false, false, 0)
 }
 
 func (s *checkSender) Distribution(metric string, value float64, hostname string, tags []string) {
-	s.sendMetricSample(metric, value, hostname, tags, metrics.DistributionType, false, false)
+	s.sendMetricSample(metric, value, hostname, tags, metrics.DistributionType, false, false, 0)
+}
+
+// GaugeWithTimestamp measures the value of a metric at a given time.
+// The value will bypass any aggregation agent side, this is useful when sending points in the past.
+func (s *checkSender) GaugeWithTimestamp(metric string, value float64, hostname string, tags []string, timestamp time.Time) error {
+	if timestamp.IsZero() || timestamp.Unix() <= 0 {
+		return fmt.Errorf("invalid timestamp")
+	}
+	s.sendMetricSample(metric, value, hostname, tags, metrics.GaugeWithTimestampType, false, false, float64(timestamp.Unix()))
+	return nil
+}
+
+// CountWithTimestamp tracks how many times something happened at the given second.
+// The value will bypass any aggregation agent side, this is useful when sending points in the past.
```

Comment:
Would it be possible to rephrase this in a way that doesn't require readers to know what aggregation does? Something like

```suggestion
// CountWithTimestamp reports a new count value to the intake with the given timestamp.
// Count time series measure how many times something happened in some time period.
// Unlike Count(), each submitted value will be passed to the intake as is, without aggregation. Each time series can have only one value per timestamp.
```